pkg/secret/api/types.go

```diff
@@ -0,0 +1,33 @@
+package api
+
+import (
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+)
+
+// Type of secret data
+type SecretType string
+
+// Valid values for SecretType
+const (
+	// TextSecretType is a secret consisting of text data
+	TextSecretType SecretType = "text"
+
+	// Base64SecretType is a secret encoded in base64
+	Base64SecretType SecretType = "base64"
+)
+
+// Secret is a resource that contains a key, password, etc to be used by pods
+type Secret struct {
+	kapi.TypeMeta   `json:",inline" yaml:",inline"`
+	kapi.ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+
+	Type SecretType `json:"type" yaml:"type"`
```

Comment:
@csrwng if there are just two options, why not call it 'Base64 bool' (false means 'plain')